vj4/model/domain.py

```diff
@@ -324,7 +324,8 @@ def get_join_settings(ddoc, now):
   await user_coll.create_index([('domain_id', 1),
                                 ('uid', 1)], unique=True)
   await user_coll.create_index([('domain_id', 1),
-                                ('display_name', 1)], unique=True)
+                                ('display_name', 1)], unique=True,
+                                partialFilterExpression={'display_name': {'$exists': True}})
```

Comment:
I see from https://docs.mongodb.com/manual/core/index-partial/ that partial index is officially recommended, but I still prefer sparse index here.

Reply:
Changed to `sparse` now.

![snipaste_2018-12-05_16-21-02](https://user-images.githubusercontent.com/5895757/49499746-e03c2680-f8a9-11e8-9e4d-d54e92bedbc1.png)

Reply:
???

Reply:
@iceb0y it seems `sparse` will cause problem

Reply:
That's because you already have the partial index. Dropping the index in mongo shell and sparse works fine:

```
db.domain.user.dropIndex('domain_id_1_display_name_1')
```